tests/manage/pv_services/pvc_snapshot/test_resource_deletion_during_snapshot_restore.py

```diff
@@ -0,0 +1,257 @@
+import logging
+import pytest
+from concurrent.futures import ThreadPoolExecutor
+
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import (
+    skipif_ocs_version, ManageTest, tier4, tier4b, ignore_leftover_label
+)
+from ocs_ci.ocs.resources.pod import cal_md5sum, verify_data_integrity
+from ocs_ci.utility.utils import ceph_health_check
+from tests import disruption_helpers
+from tests.helpers import wait_for_resource_state
+
+log = logging.getLogger(__name__)
+
+
+@tier4
+@tier4b
+@skipif_ocs_version('<4.6')
+@ignore_leftover_label(constants.drain_canary_pod_label)
+@pytest.mark.polarion_id('OCS-2369')
+class TestResourceDeletionDuringSnapshotRestore(ManageTest):
+    """
+    Tests to verify PVC snapshot and restore will succeeded if rook-ceph,
+    csi pods are re-spun while creating snapshot and while creating restore PVC
+
+    """
+    @pytest.fixture(autouse=True)
+    def setup(
+        self, project_factory, snapshot_restore_factory, create_pvcs_and_pods
+    ):
+        """
+        Create PVCs and pods
+
+        """
+        self.pvc_size = 3
+        self.pvcs, self.pods = create_pvcs_and_pods(
+            pvc_size=self.pvc_size, num_of_rbd_pvc=6, num_of_cephfs_pvc=4
+        )
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        """
+        Make sure ceph health is OK
+
+        """
+        def finalizer():
```

Comment:
this teardown seems to be redundant since we have Ceph health check done after every test case here https://github.com/red-hat-storage/ocs-ci/blob/9e04a842930d30c313fd51f6f28e1764a9512cde/tests/conftest.py#L1040

Reply:
Yes, removed finalizer.